h/api/search/test/core_test.py

```diff
@@ -0,0 +1,30 @@
+import mock
+import pytest
+
+from h.api.search import core
+
+
+search_fixtures = pytest.mark.usefixtures('query', 'models')
+
+
+@search_fixtures
+def test_search_passes_private_to_AuthFilter(query):
+    request = mock.Mock()
+
+    core.search(request, mock.Mock(), private=True)
+
+    query.AuthFilter.assert_called_once_with(request, True)
+
+
+@pytest.fixture
+def query(request):
+    patcher = mock.patch('h.api.search.core.query', autospec=True)
+    request.addfinalizer(patcher.stop)
```

Comment:
Small request. Can you make sure not to add the finalizer until the patcher is actually started? Otherwise a failure to patch (perhaps because the module moved) will also spew an error about "called stop on unstarted patcher" and the real error gets hidden further up. So:

```python
patcher = mock.patch(...)
result = patcher.start()
request.addfinalizer(patcher.stop)
return result
```